Orange/widgets/visualize/widgetutils/tree/rules.py

```diff
@@ -0,0 +1,181 @@
+class Rule:
+    """The base Rule class for tree rules."""
+
+    def merge_with(self, rule):
+        """Merge the current rule with the given rule.
+
+        Parameters
+        ----------
+        rule : Rule
+
+        Returns
+        -------
+        Rule
+
+        """
+        raise NotImplemented()
+
+
+class DiscreteRule(Rule):
+    """Discrete rule class for handling Indicator rules.
+
+    Parameters
+    ----------
+    attr_name : str
+    eq : bool
+        Should indicate whether or not the rule equals the value or not.
+    value : object
+
+    Examples
+    --------
+    Age = 30
+    >>> rule = DiscreteRule('age', True, 30)
+    Name ≠ John
+    >>> rule = DiscreteRule('name', False, 'John')
```

Comment:
The docstring example is not in reverse. The second argument of `DiscreteRule` indicates the `EQ` relation, therefore `Age is equal(True) to 30` and `Name is not equal (False) to John`.

Maybe you meant the line order, so the result should come after the example code?